src/atlas/util/Geometry.cc

```diff
@@ -45,7 +43,10 @@ void GeometrySphere::xyz2lonlat(const Point3& xyz, Point2& lonlat) const {
 /// @ref     https://en.wikipedia.org/wiki/Great-circle_navigation
 ///
 std::pair<double, double> greatCircleCourse(const Point2& lonLat1,
-                                                   const Point2& lonLat2) {
+                                            const Point2& lonLat2) {
+  if (lonLat1 == lonLat2) {
```

Comment:
This should be using bool points_equal(const Point2&, const Point2&)

I have a more consistent way to compare these in an upcoming PR (a refactoring of the geometry library) and I've already included the "course" methods there, but they will need updating to your improved version, when that time comes :-)

Reply:
That's correct, that it should use "points_equal". But that already happens. See https://github.com/ecmwf/atlas/blob/develop/src/atlas/util/Point.h#L32-L34.